pkg/plugins/crypto/api.go

```diff
@@ -0,0 +1,42 @@
+/*
```

Comment:
it seems weird that the interface definitions are under “plugins”. they are neither plugins nor implementations.

these seem to be service provider interfaces for the framework. as is vdri, storage, etc.

Reply:
suggest, at least, putting these interface folders under a `spi` directory for now (not under plugins).

perhaps:
`pkg/framework/spi/...`

or

`pkg/common/spi/...`

@rolsonquadras @fqutishat @llorllale

Reply:
sure, will move to `pkg/framework/spi/...`

Reply:
moved plugins interfaces to pkg/framework/spi/...